### PR TITLE
Improve Client Version

### PR DESCRIPTION
**Pull request by AlexSSD7**
# *Client Version* Improvements

* Better OS Names ("Windows", "macOS" instead of "Microsoft Windows", "Darwin")
* Display only OS Name. It is generally a good security practice to not disclose OS/Kernel version.

## Before:

```
# Linux
Nethermind/v1.8.61-1-9e6f95184-20200712/X64-Linux 4.19.76-linuxkit/Core3.1.5

# macOS
Nethermind/v1.8.61-1-9e6f95184-20200712/X64-Darwin Kernel Version 19.5.0: Tue May 26 20:41:44 PDT 2020; root:xnu-6153.121.2~2/RELEASE_X86_64 x86_64/Core3.1.5

# Windows
Nethermind/v1.8.61-1-9e6f95184-20200712/X64-Microsoft Windows 6.3.960/Core3.1.5
```

## After
```
# Linux
Nethermind/v1.8.61-1-9e6f95184-20200712/X64-Linux/Core3.1.5

# macOS
Nethermind/v1.8.61-1-9e6f95184-20200712/X64-macOS/Core3.1.5

# Windows
Nethermind/v1.8.61-1-9e6f95184-20200712/X64-Windows/Core3.1.5
```

### CODE REVIEW

**Comment by tkstanczak** [comment]
I am ok to accept it and if you would like to make it nonallocating then:
        private static OsPlatform GetPlatform()
        {
            if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
            {
                return OsPlatform.Windows;
            }

            if (RuntimeInformation.IsOSPlatform(OSPlatform.Linux))
            {
                return OsPlatform.Linux;
            }

            if (RuntimeInformation.IsOSPlatform(OSPlatform.OSX))
            {
                return OsPlatform.Mac;
            }

            throw new InvalidOperationException("Unsupported platform.");
        }

**Comment by AlexSSD7** [comment]
OK, I will do that. Didn't knew that there is `RuntimeInformation.IsOSPlatform()` function. 

```
throw new InvalidOperationException("Unsupported platform.");
```
This code is quite dangerous to implement as there are other operating systems like BSD, Solaris, etc. The best way to utilize both implementations is to return OS name as string and replace code above to
```
return RuntimeInformation.OSDescription.Split(" ")[0];
```
Which is equivalent to `uname` command on Unix-like operating systems.

**Comment by tkstanczak** [comment]
> OK, I will do that. Didn't know that there is `RuntimeInformation.IsOSPlatform()` function.
> 
> ```
> throw new InvalidOperationException("Unsupported platform.");
> ```
> 
> This code is quite dangerous to implement as there are other operating systems like BSD, Solaris, etc. The best way to utilize both implementations is to return OS name as string and replace code above to
> 
> ```
> return RuntimeInformation.OSDescription.Split(" ")[0];
> ```
> 
> Which is equivalent to returning the output of `uname` command on Unix-like operating systems.

Yeah, FreeBSD support was added recently and I do not think we support it out of the box. Will need to check. OSPlatform.FreeBsd is available now in this code.

**Comment by tkstanczak** [comment]
> > OK, I will do that. Didn't know that there is `RuntimeInformation.IsOSPlatform()` function.
> > ```
> > throw new InvalidOperationException("Unsupported platform.");
> > ```
> > 
> > 
> > This code is quite dangerous to implement as there are other operating systems like BSD, Solaris, etc. The best way to utilize both implementations is to return OS name as string and replace code above to
> > ```
> > return RuntimeInformation.OSDescription.Split(" ")[0];
> > ```
> > 
> > 
> > Which is equivalent to returning the output of `uname` command on Unix-like operating systems.
> 
> Yeah, FreeBSD support was added recently and I do not think we support it out of the box. Will need to check. OSPlatform.FreeBsd is available now in this code.

so actually we correctly mark the platform as unsupported / untested

**Comment by AlexSSD7** [comment]
Everything is done. OK to merge?

File: src/Nethermind/Nethermind.Core/Platform.cs [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+//  but WITHOUT ANY WARRANTY; without even the implied warranty of
+//  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+//  GNU Lesser General Public License for more details.
+// 
+//  You should have received a copy of the GNU Lesser General Public License
+//  along with the Nethermind. If not, see <http://www.gnu.org/licenses/>.
+
+using System;
+using System.Runtime.InteropServices;
+/*public class NativeLib
```
Comment (tkstanczak) [inline src/Nethermind/Nethermind.Core/Platform.cs:19]: this comment can be removed

File: src/Nethermind/Nethermind.Core/Platform.cs [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+        }
+
+        throw new InvalidOperationException("Unsupported platform.");
+    }
+}*/
+
+namespace Nethermind.Core
+{
+
+    public static class NativeLib
```
Comment (tkstanczak) [inline src/Nethermind/Nethermind.Core/Platform.cs:51]: this class can be removed

**Pull request merged**